framework/core/src/Foundation/Concerns/InteractsWithLaravel.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Foundation\Concerns;
+
+use Flarum\Locale\LocaleManager;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Str;
+
+/**
+ * Implementation of the Laravel Application Contract,
+ * for the sake of better integration with Laravel packages/ecosystem.
+ */
+trait InteractsWithLaravel
+{
+    protected array $terminatingCallbacks = [];
+    protected ?bool $isRunningInConsole = null;
+
+    public function terminating($callback): static
+    {
+        $this->terminatingCallbacks[] = $callback;
+
+        return $this;
+    }
+
+    public function terminate(): void
+    {
+        $index = 0;
+
+        while ($index < count($this->terminatingCallbacks)) {
+            $this->call($this->terminatingCallbacks[$index]);
+
+            $index++;
+        }
+    }
+
+    public function basePath($path = ''): string
+    {
+        return $this->joinPaths($this->paths->base, $path);
+    }
+
+    public function publicPath($path = ''): string
+    {
+        return $this->joinPaths($this->paths->public, $path);
+    }
+
+    public function storagePath($path = ''): string
```

Comment:
Don't we have this in `Paths`?

Reply:
yes, all these methods are implementations of the ApplicationContract, the method itself uses `Paths`